File: ios/LoverOrder/Models/Household.swift
import Foundation

// 与后端 Household 对应
struct Household: Codable, Identifiable, Hashable {
    let id: UInt
    var name: String
    var inviteCode: String
    var description: String?
    var createdBy: UInt
    var members: [AppUser]?
    var createdAt: Date?

    enum CodingKeys: String, CodingKey {
        case id
        case name
        case inviteCode = "invite_code"
        case description
        case createdBy = "created_by"
        case members
        case createdAt = "created_at"
    }
}

struct HouseholdInvite: Codable, Identifiable, Hashable {
    let id: UInt
    var householdId: UInt
    var code: String
    var expiresAt: Date?
    var maxUses: Int
    var usedCount: Int
    var isActive: Bool

    enum CodingKeys: String, CodingKey {
        case id
        case householdId = "household_id"
        case code
        case expiresAt = "expires_at"
        case maxUses = "max_uses"
        case usedCount = "used_count"
        case isActive = "is_active"
    }
}

File: backend/internal/model/household.go
package model

import (
	"time"

	"gorm.io/gorm"
)

// Household 一个"家" 支持两人/多人共用一份菜单
type Household struct {
	ID          uint           `json:"id" gorm:"primaryKey"`
	Name        string         `json:"name" gorm:"size:64;not null;default:'我们家';comment:家的名字"`
	InviteCode  string         `json:"invite_code" gorm:"uniqueIndex;size:16;not null;comment:邀请码"`
	Description string         `json:"description" gorm:"size:255;comment:简介"`
	CreatedBy   uint           `json:"created_by" gorm:"not null;index;comment:创建者"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `json:"-" gorm:"index"`

	Members []User `json:"members,omitempty" gorm:"foreignKey:HouseholdID"`
}

// TableName 一个家表名
func (Household) TableName() string {
	return "households"
}

// MemberCount 当前家庭活跃成员数
func (h *Household) MemberCount() int64 {
	if DB == nil {
		return 0
	}
	var count int64
	DB.Model(&User{}).
		Where("household_id = ? AND is_active = ?", h.ID, true).
		Count(&count)
	return count
}
